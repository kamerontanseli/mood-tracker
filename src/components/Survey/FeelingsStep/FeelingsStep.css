.FeelingsStep-title {
  text-align: center;
  margin: 50px 0;
}

.FeelingsStep-grid {
  display: flex;
  flex-direction: row;
  width: 310px;
  margin: 0 auto;
  flex-wrap: wrap;
  margin-bottom: 50px;
}

.FeelingsStep-grid-label {
  display: flex;
  width: 140px;
  height: 50px;
  align-items: center;
  justify-content: center;
  border: 2px solid #eee;
  margin: 4px;
  cursor: pointer;
  animation: all .4s ease;
}

.FeelingsStep-grid input[type="checkbox"] {
  display: none;
}

input[type="checkbox"]:checked + .FeelingsStep-grid-label {
  border-color: #8bc34a;
}

.FeelingsStep-controls {
  max-width: 200px;
  margin: 0 auto;
  display: flex;
  align-items: center;
  justify-content: space-between;
}