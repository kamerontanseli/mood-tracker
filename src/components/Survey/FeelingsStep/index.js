import "./FeelingsStep.css";
import React from "react";
import Button from "../../general/Button";
import Panel from "../../general/Panel";
import Header1 from "../../general/Header1";

export const ALL_FEELINGS = [
  "depressed",
  "optimistic",
  "bored",
  "happy",
  "woeful",
  "playful",
  "peaceful"
];

export const Feeling = ({ checked, onChange, value }) => (
  <React.Fragment>
    <input
      checked={checked}
      onChange={onChange}
      type="checkbox"
      name={value}
      id={value}
    />
    <label className="FeelingsStep-grid-label" htmlFor={value}>
      {value}
    </label>
  </React.Fragment>
);

const FeelingsStep = ({ feelings, onChange, onNext, onBack }) => (
  <Panel>
    <div className="FeelingsStep">
      <div className="FeelingsStep-title">
        <Header1>What emotions are you feeling?</Header1>
      </div>
      <div className="FeelingsStep-grid">
        {ALL_FEELINGS.map(feeling => (
          <Feeling
            checked={feelings.indexOf(feeling) > -1}
            onChange={onChange}
            value={feeling}
            key={feeling}
          />
        ))}
      </div>
      <div className="FeelingsStep-controls">
        <Button data-test="back" onClick={onBack}>
          Back
        </Button>
        <Button
          data-test="next"
          disabled={feelings.length < 1}
          onClick={onNext}
        >
          Continue
        </Button>
      </div>
    </div>
  </Panel>
);

FeelingsStep.defaultProps = {
  feelings: []
}

export default FeelingsStep;
