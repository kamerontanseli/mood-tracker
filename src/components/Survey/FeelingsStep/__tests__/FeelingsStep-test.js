import React from "react";
import { shallow, mount } from 'enzyme';
import renderer from "react-test-renderer";
import FeelingsStep, { Feeling, ALL_FEELINGS } from "..";

describe('Feeling', () => {
  it("should render", () => {
    const snapshot = renderer.create(
      <Feeling value="hello world" />
    );
    expect(snapshot).toMatchSnapshot();
  });
  it('should call onChange when input is changed', () => {
    const spy = jest.fn()
    const wrapper = mount(<Feeling onChange={spy} />)
    wrapper.find('input').first().simulate('change')
    expect(spy).toHaveBeenCalled()
  })
})

describe("render", () => {
  it("should render", () => {
    const snapshot = renderer.create(
      <FeelingsStep />
    );
    expect(snapshot).toMatchSnapshot();
  });
  it('should call onBack when [data-test="back"] is clicked', () => {
    const spy = jest.fn()
    const wrapper = mount(<FeelingsStep onBack={spy} />)
    wrapper.find('[data-test="back"]').first().simulate('click')
    expect(spy).toHaveBeenCalled()
  })
  it('should call onNext when [data-test="next"] is clicked', () => {
    const spy = jest.fn()
    const wrapper = mount(<FeelingsStep feelings={['a', 'b']} onNext={spy} />)
    wrapper.find('[data-test="next"]').first().simulate('click')
    expect(spy).toHaveBeenCalled()
  })
  it('should render a Feeling for each feeling in ALL_FEELINGS', () => {
    const wrapper = mount(<FeelingsStep />)
    expect(wrapper.find(Feeling).length).toBe(ALL_FEELINGS.length)
  })
});
