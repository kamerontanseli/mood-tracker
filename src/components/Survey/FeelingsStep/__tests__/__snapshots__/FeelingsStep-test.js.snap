// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feeling should render 1`] = `
Array [
  <input
    id="hello world"
    name="hello world"
    type="checkbox"
  />,
  <label
    className="FeelingsStep-grid-label"
    htmlFor="hello world"
  >
    hello world
  </label>,
]
`;

exports[`render should render 1`] = `
<div
  className="Panel"
>
  <div
    className="FeelingsStep"
  >
    <div
      className="FeelingsStep-title"
    >
      <h1
        className="Header1"
      >
        What emotions are you feeling?
      </h1>
    </div>
    <div
      className="FeelingsStep-grid"
    >
      <input
        checked={false}
        id="depressed"
        name="depressed"
        type="checkbox"
      />
      <label
        className="FeelingsStep-grid-label"
        htmlFor="depressed"
      >
        depressed
      </label>
      <input
        checked={false}
        id="optimistic"
        name="optimistic"
        type="checkbox"
      />
      <label
        className="FeelingsStep-grid-label"
        htmlFor="optimistic"
      >
        optimistic
      </label>
      <input
        checked={false}
        id="bored"
        name="bored"
        type="checkbox"
      />
      <label
        className="FeelingsStep-grid-label"
        htmlFor="bored"
      >
        bored
      </label>
      <input
        checked={false}
        id="happy"
        name="happy"
        type="checkbox"
      />
      <label
        className="FeelingsStep-grid-label"
        htmlFor="happy"
      >
        happy
      </label>
      <input
        checked={false}
        id="woeful"
        name="woeful"
        type="checkbox"
      />
      <label
        className="FeelingsStep-grid-label"
        htmlFor="woeful"
      >
        woeful
      </label>
      <input
        checked={false}
        id="playful"
        name="playful"
        type="checkbox"
      />
      <label
        className="FeelingsStep-grid-label"
        htmlFor="playful"
      >
        playful
      </label>
      <input
        checked={false}
        id="peaceful"
        name="peaceful"
        type="checkbox"
      />
      <label
        className="FeelingsStep-grid-label"
        htmlFor="peaceful"
      >
        peaceful
      </label>
    </div>
    <div
      className="FeelingsStep-controls"
    >
      <button
        className="Button"
        data-test="back"
      >
        Back
      </button>
      <button
        className="Button"
        data-test="next"
        disabled={true}
      >
        Continue
      </button>
    </div>
  </div>
</div>
`;
